src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,622 @@
+
+        if tracing::enabled!(tracing::Level::TRACE) {
+            span.record(
+                "merged_nix_config",
+                &self
+                    .nix_configs
+                    .merged_nix_config
+                    .settings()
+                    .iter()
+                    .map(|(k, v)| format!("{k}='{v}'"))
+                    .collect::<Vec<_>>()
+                    .join(" "),
+            );
+        }
+        span
+    }
+
+    fn execute_description(&self) -> Vec<ActionDescription> {
+        vec![ActionDescription::new(
+            self.tracing_synopsis(),
+            vec![format!(
+                "If {} already exists, we will attempt to merge the current settings with our settings; \
+                otherwise, it will be created with only our settings",
+                self.path.display()
+            )],
+        )]
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn execute(&mut self) -> Result<(), ActionError> {
+        let Self { path, nix_configs } = self;
+
+        if tracing::enabled!(tracing::Level::TRACE) {
+            let span = tracing::Span::current();
+            span.record(
+                "merged_nix_config",
+                nix_configs
+                    .merged_nix_config
+                    .settings()
+                    .iter()
+                    .map(|(k, v)| format!("{k}='{v}'"))
+                    .collect::<Vec<_>>()
+                    .join(" "),
+            );
+        }
+
+        // Create a temporary file in the same directory as the one
+        // that the final file goes in, so that we can rename it
+        // atomically
+        let parent_dir = path.parent().expect("File must be in a directory");
+        let mut temp_file_path = parent_dir.to_owned();
+        {
+            let mut rng = rand::thread_rng();
+            temp_file_path.push(format!("nix-installer-tmp.{}", rng.gen::<u32>()));
+        }
+        let mut temp_file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            // If the file is created, ensure that it has harmless
+            // permissions regardless of whether the mode will be
+            // changed later (if we ever create setuid executables,
+            // they should only become setuid once they are owned by
+            // the appropriate user)
+            .mode(0o600)
+            .open(&temp_file_path)
+            .await
+            .map_err(|e| {
+                ActionError::Open(temp_file_path.clone(), e)
+            })?;
+
+        let mut new_config = String::new();
+
+        if let Some(existing_nix_config) = nix_configs.existing_nix_config.as_mut() {
+            let discovered_buf = tokio::fs::read_to_string(&path)
+                .await
+                .map_err(|e| ActionError::Read(path.to_path_buf(), e))?;
+
+            // FIXME: ugly, but it does what I need...
+            let (associated_lines, _, _) = discovered_buf.split('\n').fold(
```

Comment:
Is there a reason you prefer `split()` here to `lines()`?

Reply:
Uh, not really. I think I was not using a `String` at one point, and it had `split` but not `lines`, but that's obviously no longer the case.